src/datasets/dataset_dict.py

```diff
@@ -1285,26 +1285,35 @@ def load_from_disk(
         fs_token_paths = fsspec.get_fs_token_paths(dataset_dict_path, storage_options=storage_options)
         fs: fsspec.AbstractFileSystem = fs_token_paths[0]
 
-        dataset_dict = DatasetDict()
         if is_remote_filesystem(fs):
             dest_dataset_dict_path = extract_path_from_uri(dataset_dict_path)
+            path_join = posixpath.join
         else:
             fs = fsspec.filesystem("file")
             dest_dataset_dict_path = dataset_dict_path
-        dataset_dict_json_path = Path(dest_dataset_dict_path, config.DATASETDICT_JSON_FILENAME).as_posix()
-        dataset_info_path = Path(dest_dataset_dict_path, config.DATASET_INFO_FILENAME).as_posix()
-        if fs.isfile(dataset_info_path) and not fs.isfile(dataset_dict_json_path):
+            path_join = os.path.join
+
+        dataset_dict_json_path = path_join(dest_dataset_dict_path, config.DATASETDICT_JSON_FILENAME)
+        dataset_state_json_path = path_join(dest_dataset_dict_path, config.DATASET_STATE_JSON_FILENAME)
+        dataset_info_path = path_join(dest_dataset_dict_path, config.DATASET_INFO_FILENAME)
+        if not fs.isfile(dataset_dict_json_path):
+            if fs.isfile(dataset_info_path) and fs.isfile(dataset_state_json_path):
+                raise FileNotFoundError(
+                    f"No such file or directory: '{dataset_dict_json_path}'. Expected to load a `DatasetDict` object, but got a `Dataset`. Please use either `datasets.load_from_disk` or `Dataset.load_from_disk` instead."
+                )
             raise FileNotFoundError(
-                f"No such file or directory: '{dataset_dict_json_path}'. Expected to load a DatasetDict object, but got a Dataset. Please use datasets.load_from_disk instead."
+                f"No such file or directory: '{dataset_dict_json_path}'. Expected to load a `DatasetDict` object, but provided path is neither a `DatasetDict` nor a `Dataset`"
```

Comment:
I included that extra information basically to let the user know that it's not because it's a `Dataset` instead of ` DatasetDict`, as with that if-statement we can also guide the user, I thought that specifying that it's neither of those was useful!

P.S. Same thing applies to the `FileNotFound` error you mentioned in the other suggestion below in `src/datasets/arrow_dataset.py`

Reply:
Here we're talking about `DatasetDict.load_from_disk` - I don't think we need to tell the user that or they may think it's also fine to provide a `Dataset` directory, which is not correct

Reply:
True @lhoestq, ok then I'll commit this separately to join this suggestion and the one below in a single commit :)